pkg/cmd/pulumi/config.go

```diff
@@ -80,10 +80,165 @@ func newConfigCmd() *cobra.Command {
+				return copySingleConfigKey(args[0], path, currentStack, currentProjectStack, destinationStack,
+					destinationProjectStack)
+			}
+
+			return copyEntireConfigMap(currentStack, currentProjectStack, destinationStack, destinationProjectStack)
+		}),
+	}
+
+	cpCommand.PersistentFlags().BoolVar(
+		&path, "path", false,
+		"The key contains a path to a property in a map or list to set")
+	cpCommand.PersistentFlags().StringVarP(
+		&destinationStackName, "new-stack", "", "",
+		"The name of the new stack to copy the config to")
+
+	return cpCommand
+}
+
+func copySingleConfigKey(configKey string, path bool, currentStack backend.Stack,
+	currentProjectStack *workspace.ProjectStack, destinationStack backend.Stack,
+	destinationProjectStack *workspace.ProjectStack) error {
+	var decrypter config.Decrypter
+	key, err := parseConfigKey(configKey)
+	if err != nil {
+		return errors.Wrap(err, "invalid configuration key")
+	}
+
+	v, ok, err := currentProjectStack.Config.Get(key, path)
+	if err != nil {
+		return err
+	}
+	if ok {
+		if v.Secure() {
+			var err error
+			if decrypter, err = getStackDecrypter(currentStack); err != nil {
+				return errors.Wrap(err, "could not create a decrypter")
+			}
+		} else {
+			decrypter = config.NewPanicCrypter()
+		}
+
+		raw, err := v.Value(decrypter)
+		if err != nil {
+			return errors.Wrap(err, "could not decrypt configuration value")
+		}
+
+		// now we start the work to copy it - if it was encrypted before then we need to do the same again
+		if v.Secure() {
+			// it was encrypted before, now we need to re-encrypt
+			c, cerr := getStackEncrypter(destinationStack)
+			if cerr != nil {
+				return cerr
+			}
+			enc, eerr := c.EncryptValue(raw)
+			if eerr != nil {
+				return eerr
+			}
+			v = config.NewSecureValue(enc)
+		} else {
+			v = config.NewValue(raw)
+		}
+
+		err = destinationProjectStack.Config.Set(key, v, path)
+		if err != nil {
+			return err
+		}
+
+		return saveProjectStack(destinationStack, destinationProjectStack)
+	}
+
+	return nil
+}
+
+func copyEntireConfigMap(currentStack backend.Stack,
+	currentProjectStack *workspace.ProjectStack, destinationStack backend.Stack,
+	destinationProjectStack *workspace.ProjectStack) error {
+
+	var decrypter config.Decrypter
+	currentConfig := currentProjectStack.Config
+	if currentConfig.HasSecureValue() {
+		dec, decerr := getStackDecrypter(currentStack)
+		if decerr != nil {
+			return decerr
+		}
+		decrypter = dec
+	} else {
+		decrypter = config.NewPanicCrypter()
+	}
+
+	encrypter, cerr := getStackEncrypter(destinationStack)
+	if cerr != nil {
+		return cerr
+	}
+
+	newProjectConfig, err := currentConfig.Copy(decrypter, encrypter)
+	if err != nil {
+		return err
+	}
+
+	for key, val := range newProjectConfig {
+		err = destinationProjectStack.Config.Set(key, val, false)
+		if err != nil {
+			return err
+		}
+
+		err = saveProjectStack(destinationStack, destinationProjectStack)
+		if err != nil {
+			return err
+		}
```

Comment:
Any reason why we're saving on each iteration of the loop? Should we save once after the loop?

Reply:
This was to ensure that if an error occurred then at least we will have saved some of the config - I guess it's more efficient to do it once - changing now